server/isomorphic-routing/index.js

```diff
@@ -29,6 +36,7 @@ export function serverRouter( expressApp, setUpRoute, section ) {
 				setUpRoute,
 				combineMiddlewares(
 					setSectionMiddlewareFactory( section ),
+					setRouteMiddleware,
```

Comment:
This change caused https://github.com/Automattic/wp-calypso/issues/18919

Reply:
What do we actually need this line for? :slightly_smiling_face:

Reply:
In this case it's used to set `state.ui.oauth2Clients.currentClientId` from our `client_id` query parameter.